File: app/src/main/java/com/rakib/foodbankbd/adapters/OfferRecyclerViewAdapter.java
package com.rakib.foodbankbd.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.rakib.foodbankbd.OfferActivity;
import com.rakib.foodbankbd.R;
import com.rakib.foodbankbd.entities.OfferObject;

import java.util.List;

public class OfferRecyclerViewAdapter extends RecyclerView.Adapter<OfferRecyclerViewHolder> {
    private Context context;
    private List<OfferObject> allOffers;

    public  OfferRecyclerViewAdapter(Context context, List<OfferObject> allOffers) {
        this.context = context;
        this.allOffers = allOffers;
    }

    @NonNull
    @Override
    public OfferRecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View layoutView = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_area, parent, false);
        OfferRecyclerViewHolder offerHolder = new OfferRecyclerViewHolder(layoutView);
        return offerHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull OfferRecyclerViewHolder holder, final int position) {
        final OfferObject singleOffer = allOffers.get(position);

        holder.offerName.setText(singleOffer.getOfferName());
        holder.offerImage.setImageResource(singleOffer.getOfferImage());

        holder.offerImage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent offerIntent = new Intent(context, OfferActivity.class);

                GsonBuilder gsonBuilder = new GsonBuilder();
                Gson gson = gsonBuilder.create();

                String stringObjectRepresentation = gson.toJson(singleOffer);

                offerIntent.putExtra("OFFER_CODE", stringObjectRepresentation);
                context.startActivity(offerIntent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return allOffers.size();
    }
}

File: app/src/main/java/com/rakib/foodbankbd/adapters/OfferRecyclerViewAdapter.kt
package com.rakib.foodbankbd.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.google.gson.GsonBuilder
import com.rakib.foodbankbd.OfferActivity
import com.rakib.foodbankbd.R
import com.rakib.foodbankbd.entities.OfferObject

class OfferRecyclerViewAdapter(private val context: Context, private val allOffers: List<OfferObject>) : RecyclerView.Adapter<OfferRecyclerViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OfferRecyclerViewHolder {
        val layoutView = LayoutInflater.from(parent.context).inflate(R.layout.layout_area, parent, false)
        return OfferRecyclerViewHolder(layoutView)
    }

    override fun onBindViewHolder(holder: OfferRecyclerViewHolder, position: Int) {
        val singleOffer = allOffers[position]
        holder.offerName.text = singleOffer.offerName
        holder.offerImage.setImageResource(singleOffer.offerImage)

        holder.offerImage.setOnClickListener {
            val offerIntent = Intent(context, OfferActivity::class.java)
            val gsonBuilder = GsonBuilder()
            val gson = gsonBuilder.create()
            val stringObjectRepresentation = gson.toJson(singleOffer)
            offerIntent.putExtra("OFFER_CODE", stringObjectRepresentation)
            context.startActivity(offerIntent)
        }
    }

    override fun getItemCount(): Int {
        return allOffers.size
    }
}